pkg/controller/gardenletdeployer/actuator_test.go

```diff
@@ -529,7 +533,73 @@ var _ = Describe("Interface", func() {
 				))
 			})
 
+			It("should return error when DoNotCopyBackupCredentials feature gate is enabled and backup secret does not exist", func() {
+				DeferCleanup(test.WithFeatureGate(features.DefaultFeatureGate, features.DoNotCopyBackupCredentials, true))
+
+				expectCheckSeedSpec()
+				recorder.EXPECT().Eventf(managedSeed, corev1.EventTypeNormal, gardencorev1beta1.EventReconciling, "Ensuring garden namespace in target cluster")
+				expectCreateGardenNamespace()
+				recorder.EXPECT().Event(managedSeed, corev1.EventTypeNormal, gardencorev1beta1.EventReconciling, "Reconciling seed secrets")
+
+				gardenClient.EXPECT().Get(ctx, client.ObjectKey{Namespace: namespace, Name: backupSecretName}, gomock.AssignableToTypeOf(&corev1.Secret{})).DoAndReturn(
+					func(_ context.Context, _ client.ObjectKey, _ *corev1.Secret, _ ...client.GetOption) error {
+						return apierrors.NewNotFound(corev1.Resource("secret"), backupSecretName)
+					},
+				)
```

Comment:
It can be simplified as 
```suggestion
				gardenClient.EXPECT().Get(ctx, client.ObjectKey{Namespace: namespace, Name: backupSecretName}, gomock.AssignableToTypeOf(&corev1.Secret{})).Return(apierrors.NewNotFound(corev1.Resource("secret"), backupSecretName))
```

as there is nothing custom that is being executed in the `DoAndReturn` - we simply return err.